apps/cowswap-frontend/src/modules/affiliate/state/ReferralContext.tsx

```diff
@@ -0,0 +1,386 @@
+import {
+  ReactNode,
+  createContext,
+  useCallback,
+  useContext,
+  useEffect,
+  useMemo,
+  useState,
+  Dispatch,
+  SetStateAction,
+} from 'react'
+
+import { isProdLike } from '@cowprotocol/common-utils'
+
+import { REFERRAL_STORAGE_KEY } from '../constants'
+import {
+  ReferralActions,
+  ReferralContextValue,
+  ReferralDomainState,
+  ReferralModalSource,
+  ReferralVerificationStatus,
+  WalletReferralState,
+} from '../types'
+import { sanitizeReferralCode } from '../utils/code'
+
+export interface ReferralProviderProps {
+  children: ReactNode
+}
+
+const initialState: ReferralDomainState = {
+  modalOpen: false,
+  modalSource: null,
+  editMode: false,
+  inputCode: '',
+  savedCode: undefined,
+  incomingCode: undefined,
+  verification: { kind: 'idle' },
+  wallet: { status: 'unknown' },
+  shouldAutoVerify: false,
+  lastVerificationRequest: undefined,
+  pendingVerificationRequest: undefined,
+}
+
+const noop = (): void => undefined
+
+const noopActions: ReferralActions = {
+  openModal: noop,
+  closeModal: noop,
+  setInputCode: noop,
+  enableEditMode: noop,
+  disableEditMode: noop,
+  saveCode: noop,
+  removeCode: noop,
+  setIncomingCode: noop,
+  setWalletState: noop,
+  startVerification: noop,
+  completeVerification: noop,
+  setShouldAutoVerify: noop,
+  setSavedCode: noop,
+  requestVerification: noop,
+  clearPendingVerification: noop,
+}
+
+const ReferralContext = createContext<ReferralContextValue>({ ...initialState, actions: noopActions })
```

Comment:
In general I'm not opposed to native React contexts, but we commonly use jotai atoms instead. Is there a reson to use a context instead of atom?